[DOC] Add ingest processors documentation (#4299)
opensearch-project/documentation-website: _api-reference/ingest-apis/processors-reference/lowercase.md

---
layout: default
title: Lowercase
parent: Ingest processors 
grand_parent: Ingest APIs
nav_order: 210
---

# Lowercase

This processor converts all the text in a specific field to lowercase letters. The syntax for the `lowercase` processor is: 

```json
{
  "lowercase": {
    "field": "field_name"
  }
}
```

#### Configuration parameters

The following table lists the required and optional parameters for the `lowercase` processor.

| Name  | Required  | Description  |
|---|---|---|
| `field`  | Required  | Specifies the name of the field that you want to remove. |
| `target_field`  | Optional  | Specifies the name of the field to store the converted value in. Default is `field`. By default, `field` is updated in-place. |
| `ignore_missing`  | Optional  | Specifies whether the processor should ignore documents that do not have the specified field. Default is `false`.  |
| `ignore_failure`  | Optional  |  Specifies whether the processor should continue processing documents even if it fails to remove the specified field. Default is `false`.  |
| `on_failure`  | Optional  | Defines the processors to be deployed immediately following the failed processor.  |

> | `if`  | Optional  | Conditionally deploys the processor based on the value of the field. The `value` parameter specifies the value that you want to compare the field to. |

Reviewer: ```suggestion
| `if`  | Optional  | Conditional expression that determines whether the processor should be deployed.  |
```
